File: research/survey/wasm-field-2026-08/src/go/recursion.go
package main

import "fmt"

func run(n int64, acc int64) int64 {
	if n <= 0 {
		return acc
	}
	return run(n-1, (acc+n)%999983)
}

func main() {
	var r int64 = 0
	var acc int64 = 0
	for r < 30 {
		acc = (acc + run(1000000, 0)) % 999983
		r++
	}
	fmt.Println(acc)
}

File: research/survey/wasm-field-2026-08/src/rust/recursion.rs
fn go(n: i64, acc: i64) -> i64 {
    if n <= 0 { acc } else { go(n - 1, (acc + n) % 999983) }
}

fn main() {
    let mut r: i64 = 0;
    let mut acc: i64 = 0;
    while r < 30 {
        acc = (acc + go(1000000, 0)) % 999983;
        r += 1;
    }
    println!("{}", acc);
}
